hypershift-operator/controllers/scheduler/placeholders.go

```diff
@@ -319,35 +323,37 @@ func (r *placeholderUpdater) reconcile(
 		return true, nil, nil
 	}
 
-	nodes, err := r.listNodes(ctx, client.HasLabels{HostedClusterNameLabel, OSDFleetManagerPairedNodesLabel})
+	pairLabelsAssignedToHostedClusters := sets.Set[string]{}
+	assignedConfigMaps, err := r.listConfigMaps(ctx, client.HasLabels{pairLabelKey}, client.InNamespace(placeholderNamespace))
 	if err != nil {
 		return false, nil, err
 	}
-
-	pairedNodesAssignedToHostedClusters := sets.Set[string]{}
-	for _, node := range nodes.Items {
-		pairedNodesAssignedToHostedClusters.Insert(node.ObjectMeta.Labels[OSDFleetManagerPairedNodesLabel])
+	for _, cm := range assignedConfigMaps.Items {
+		pairLabel := cm.Labels[pairLabelKey]
+		if pairLabel != "" {
+			pairLabelsAssignedToHostedClusters.Insert(pairLabel)
+		}
 	}
 
-	existingPairedNodes := sets.Set[string]{}
+	existingPairLabels := sets.Set[string]{}
 	if affinity := deployment.Spec.Template.Spec.Affinity; affinity != nil {
 		if nodeAffinity := affinity.NodeAffinity; nodeAffinity != nil {
 			if selector := nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution; selector != nil {
 				for _, term := range selector.NodeSelectorTerms {
 					for _, expression := range term.MatchExpressions {
 						if expression.Key == OSDFleetManagerPairedNodesLabel && expression.Operator == corev1.NodeSelectorOpNotIn {
-							existingPairedNodes.Insert(expression.Values...)
+							existingPairLabels.Insert(expression.Values...)
 						}
 					}
 				}
 			}
 		}
 	}
 
-	if !pairedNodesAssignedToHostedClusters.Equal(existingPairedNodes) {
+	if !pairLabelsAssignedToHostedClusters.Equal(existingPairLabels) {
 		// the set of paired nodes on which hosted clusters have been scheduled has changed, we need to update the
 		// node affinity configuration on the deployment
-		return false, newDeployment(placeholderNamespace, placeholderSize, parsedIndex, pairedNodesAssignedToHostedClusters.UnsortedList()), nil
+		return false, newDeployment(placeholderNamespace, placeholderSize, parsedIndex, pairLabelsAssignedToHostedClusters.UnsortedList()), nil
```

Comment:
nit: using `pairLabelsAssignedToHostedClusters` and `pairedNodes` in `newDeployment` might be a bit confusing.